src/linear-client.ts

```diff
@@ -0,0 +1,351 @@
+  switchMode(params?: any): GenericAPIResponse {
+    assert(params, 'No params passed');
+    assert(params.symbol, 'Parameter symbol is required');
+    assert(params.tp_sl_mode, 'Parameter tp_sl_mode is required');
+
+    return this.requestWrapper.post('private/linear/tpsl/switch-mode', params)
+  }
+
+  /**
+   * @public Add or reduce margin.
+   */
+  addMargin(params?: any): GenericAPIResponse {
+    assert(params, 'No params passed');
+    assert(params.symbol, 'Parameter symbol is required');
+    assert(params.side, 'Parameter side is required');
+
+    return this.requestWrapper.post('private/linear/position/add-margin', params)
+  }
+
+  /**
+   * @public Set leverage.
+   */
+  setLeverage(params?: any): GenericAPIResponse {
+    assert(params, 'No params passed');
+    assert(params.symbol, 'Parameter symbol is required');
+    assert(params.buy_leverage, 'Parameter buy_leverage is required');
+    assert(params.sell_leverage, 'Parameter sell_leverage is required');
+
+    return this.requestWrapper.post('private/linear/position/set-leverage', params)
+  }
+
+  /**
+   * @public Set trading stop.
+   */
+  setTradingStop(params?: any): GenericAPIResponse {
+    assert(params, 'No params passed');
+    assert(params.symbol, 'Parameter symbol is required');
+    assert(params.side, 'Parameter side is required');
+
+    return this.requestWrapper.post('private/linear/position/trading-stop', params);
+  }
+
+  /**
+   * @public Get list of executed trades.
+   */
+  getTradeRecords(params?: any): GenericAPIResponse {
+    assert(params, 'No params passed');
+    assert(params.symbol, 'Parameter symbol is required');
+
+    return this.requestWrapper.get('private/linear/trade/execution/list', params)
+  }
+
+  /**
+   * @public Get closed PnL.
+   */
+  getClosedPNL(params?: any): GenericAPIResponse {
+    assert(params, 'No params passed');
+    assert(params.symbol, 'Parameter symbol is required');
+
+    return this.requestWrapper.get('private/linear/trade/closed-pnl/list', params)
+  }
+
+  /**
+   * @public Get risk limit.
+   */
+  getRiskLimitList(params?: any): GenericAPIResponse {
+    assert(params, 'No params passed');
+    assert(params.symbol, 'Parameter symbol is required');
+
+    return this.requestWrapper.get('public/linear/risk-limit');
+  }
+
+  /**
+   * @public Get the last funding fee.
+   */
+  getMyLastFundingFee(params: any): GenericAPIResponse {
+    assert(params, 'No params passed');
+    assert(params.symbol, 'Parameter symbol is required');
+
+    return this.requestWrapper.get('private/linear/funding/prev-funding', params);
+  }
+
+  /**
+   * @public Get the next predicted funding.
+   */
+  getPredictedFunding(params: any): GenericAPIResponse {
+    assert(params, 'No params passed');
+    assert(params.symbol, 'Parameter symbol is required');
+
+    return this.requestWrapper.get('private/linear/funding/predicted-funding', params);
+  }
+
+}
+
+export const LinearClient = SharedEndpoints(_LinearClient);
```

Comment:
Why not the reverse? 
```export class LinearClient extends SharedEndpoints { ... implementation ... }``` 

It would make SharedEndpoints a lot lighter too.